pkg/providers/vsphere/vmlifecycle/bootstrap_linuxprep.go

```diff
@@ -35,15 +38,43 @@ func BootStrapLinuxPrep(
 		return nil, nil, fmt.Errorf("failed to create GOSC NIC mappings: %w", err)
 	}
 
-	customSpec := &vimtypes.CustomizationSpec{
-		Identity: &vimtypes.CustomizationLinuxPrep{
-			HostName: &vimtypes.CustomizationFixedName{
-				Name: bsArgs.HostName,
-			},
-			Domain:     bsArgs.DomainName,
-			TimeZone:   linuxPrepSpec.TimeZone,
-			HwClockUTC: linuxPrepSpec.HardwareClockIsUTC,
+	identity := &vimtypes.CustomizationLinuxPrep{
+		HostName: &vimtypes.CustomizationFixedName{
+			Name: bsArgs.HostName,
 		},
+		Domain:     bsArgs.DomainName,
+		TimeZone:   linuxPrepSpec.TimeZone,
+		HwClockUTC: linuxPrepSpec.HardwareClockIsUTC,
+	}
+
+	if pkgcfg.FromContext(vmCtx).Features.GuestCustomizationVCDParity {
+		if linuxPrepSpec.ExpirePasswordAfterNextLogin {
+			identity.ResetPassword = vimtypes.NewBool(true)
+		}
+
+		if bsArgs.LinuxPrep != nil {
+			if bsArgs.LinuxPrep.Password != "" {
+				identity.Password = &vimtypes.CustomizationPassword{
+					Value:     bsArgs.LinuxPrep.Password,
+					PlainText: true,
+				}
+			}
+
+			identity.ScriptText = bsArgs.LinuxPrep.ScriptText
+		}
+
+		if id, ok := vmCtx.VM.Annotations[pkgconst.VCFAIDAnnotationKey]; ok {
```

Comment:
Please rewrite as `if id := vmCtx.VM.Annotations[pkgconst.VCFAIDAnnotationKey]; id != "" {`. Unlike real ExtraConfig, there is no real reason to allow an empty value to clear anything.